### ISSUE TITLE
Fix -Warray-bounds warnings in kernel/bpf/lpm_trie.c:

### ISSUE THREAD

**Issue by GustavoARSilva** [KSPP/linux#301]
```c
kernel/bpf/lpm_trie.c:207:51: warning: array subscript i is outside array bounds of ‘const __u8[0]’ {aka ‘const unsigned char[]’} [-Warray-bounds=]
kernel/bpf/lpm_trie.c:218:63: warning: array subscript i is outside array bounds of ‘const __u8[0]’ {aka ‘const unsigned char[]’} [-Warray-bounds=]
kernel/bpf/lpm_trie.c:195:51: warning: array subscript i is outside array bounds of ‘const __u8[0]’ {aka ‘const unsigned char[]’} [-Warray-bounds=]
```

**Comment by GustavoARSilva** [comment]
Patch: https://lore.kernel.org/linux-hardening/20230204183241.never.481-kees@kernel.org/

**Comment by kees** [comment]
Bulk fixes landed in commit 94dfc73e7cf4a31da66b8843f0b9283ddd6b8381.
